support/ebpf/bpfdefs.h

```diff
@@ -103,39 +105,40 @@ static long (*bpf_probe_read_kernel)(void *dst, int size, const void *unsafe_ptr
+    })
+
+  // Sends `SIGTRAP` to the current task, killing it and capturing a coredump.
+  //
+  // Only use this in code paths that you expect to be hit by a very specific process that you
+  // intend to debug. Placing it into frequently taken code paths might otherwise take down
+  // important system processes like sshd or your window manager. For frequently taken cases,
+  // prefer using the `DEBUG_CAPTURE_COREDUMP_IF_TGID` macro.
+  //
+  // This macro requires linking against kernel headers >= 5.6.
+  #define DEBUG_CAPTURE_COREDUMP()                                                                 \
+    ({                                                                                             \
+      if (with_debug_output) {                                                                     \
         /* We don't define `bpf_send_signal_thread` globally because it requires a      */         \
         /* rather recent kernel (>= 5.6) and otherwise breaks builds of older versions. */         \
         long (*bpf_send_signal_thread)(u32 sig) = (void *)BPF_FUNC_send_signal_thread;             \
         bpf_send_signal_thread(SIGTRAP);                                                           \
-      })
-
-    // Like `DEBUG_CAPTURE_COREDUMP`, but only coredumps if the current task is a member of the
-    // given thread group ID ("process").
-    #define DEBUG_CAPTURE_COREDUMP_IF_TGID(tgid)                                                   \
-      ({                                                                                           \
-        if (bpf_get_current_pid_tgid() >> 32 == (tgid)) {                                          \
-          DEBUG_PRINT("coredumping process %d", (tgid));                                           \
-          DEBUG_CAPTURE_COREDUMP();                                                                \
-        }                                                                                          \
-      })
-  #else
-    #define DEBUG_PRINT(fmt, ...)
-    #define DEBUG_CAPTURE_COREDUMP()
-    #define DEBUG_CAPTURE_COREDUMP_IF_TGID(tgid)
-  #endif
+      }                                                                                            \
+    })
+
+  // Like `DEBUG_CAPTURE_COREDUMP`, but only coredumps if the current task is a member of the
+  // given thread group ID ("process").
+  #define DEBUG_CAPTURE_COREDUMP_IF_TGID(tgid)                                                     \
+    ({                                                                                             \
+      if (with_debug_output && bpf_get_current_pid_tgid() >> 32 == (tgid)) {                       \
+        DEBUG_PRINT("coredumping process %d", (tgid));                                             \
+        DEBUG_CAPTURE_COREDUMP();                                                                  \
```

Comment:
Or split these two lines out into a separate macro `CAPTURE_COREDUMP` and have this place and `DEBUG_CAPTURE_COREDUMP` call it.

```suggestion
        long (*bpf_send_signal_thread)(u32 sig) = (void *)BPF_FUNC_send_signal_thread;             \
        bpf_send_signal_thread(SIGTRAP);                                                                  \
```